### PR TITLE
feat(desktop): make pull request reviews actionable

### PR DESCRIPTION
**Pull request by thomaspblock**
## Summary

Pull request reviews now keep the decision and its explanation together: reviewers can comment or request changes from the same composer, approve with an optional summary, and jump from inline activity directly to the referenced diff line. Review activity is presented as a chronological timeline that starts with the latest three entries, remembers expansion choices, and keeps available lifecycle actions visible when fully collapsed.

Repository owners and their managed agents can also move pull requests between draft, open, and closed states, while source-channel metadata links the review back to the conversation that produced it. Reviewer selection, author identity, timestamps, profile previews, and project navigation have been polished to make the full review flow easier to scan and operate.

## Validation

- `pnpm --dir desktop typecheck`
- Pull request review Playwright workflow, including comment-backed decisions, inline diff navigation, lifecycle controls, and remembered history state
- Pre-push desktop checks and test suite
- Full desktop smoke run: 663 passed and 1 skipped; the branch-related inline-review expectation was corrected and passed twice in targeted reruns. One unrelated video-attachment scenario remains flaky at varying checkpoints.

## Test plan

- [x] Request and remove reviewers through the centered dialog
- [x] Submit comments, request changes, and approve with a summary
- [x] Open inline review activity at the referenced diff line
- [x] Collapse, reopen, and fully expand review history
- [x] Exercise draft, close, reopen, and merge controls
- [x] Verify source-channel navigation and granular project timestamps

### CODE REVIEW

**Comment by thomaspblock** [comment]
🤖 ## Screenshots

### Add reviewer
The reviewer picker now opens as a centered, searchable dialog.

00-add-reviewer-dialog

### Review requested
Reviewer activity appears in the chronological review timeline.

01-review-requested

### Changes requested
A change request carries the reviewer's explanation and warning treatment.

05-changes-requested

### Approved
Approval summaries live in the same review activity timeline.

02-approved

### Draft lifecycle
Owners and managed agents can move the pull request through its lifecycle controls.

03-draft

### Inline review
Inline requests include a clickable code reference that opens the relevant diff line.

04-inline-comment

File: desktop/src/features/projects/hooks.ts [hunk 1]
```diff
@@ -484,6 +489,12 @@ async function createProjectPullRequestComment({
           ["line", String(normalizedAnchor.line)],
         ]
       : []),
+    ...(decision
+      ? [
+          ["t", PR_CHANGES_REQUESTED_LABEL],
+          ...(!normalizedAnchor ? [["c", pullRequest.commit as string]] : []),
```
Comment (wpfleger96) [inline desktop/src/features/projects/hooks.ts:496]: The approve path (`PullRequestReviewCard.runReviewDecision`) allocates its timestamp through `nextProjectPullRequestReviewCreatedAt()` so it always advances past the latest decision, but this request-changes path signs with plain `now`. If the same reviewer lands two opposite decisions in the same second, both events tie on `created_at` and `reviewDecisionsForPullRequest` falls back to event-id lexicographic order (`projectPullRequests.mjs:267-271`) — so the later decision can lose. Worst case the UI toasts "Changes requested" while the PR effectively stays approved. Rare for humans given dialog latency, but plausible for agents driving the UI, and the newly added approval -> request-changes direction has no E2E covering it (the old "rapid opposite decisions" test was removed in this PR).

I think the fix is to thread `nextProjectPullRequestReviewCreatedAt(pullRequest, now)` in as `createdAt` here when `decision` is set, so all review decisions (approve + top-level/inline change requests) share one PR-scoped monotonic path — plus a regression test for the same-second approve -> request-changes case.

File: crates/buzz-sdk/src/builders.rs [hunk 2]
```diff
@@ -1352,6 +1354,14 @@ pub fn build_git_pull_request(
         tags.push(tag(&["t", label])?);
     }
     tags.push(tag(&["c", &meta.commit])?);
+    if let Some(ref channel_id) = meta.channel_id {
+        if channel_id.trim().is_empty() {
+            return Err(SdkError::InvalidInput(
+                "channel_id must not be empty".into(),
+            ));
+        }
+        tags.push(tag(&["h", channel_id])?);
```
Comment (wpfleger96) [inline crates/buzz-sdk/src/builders.rs]: This only rejects an empty `channel_id` and then emits the original untrimmed string, but `--channel` is documented as a channel UUID and the desktop navigates directly to it. Worth parsing it as a UUID (consistent with the rest of the SDK) or at least trimming/validating shape before adding the `h` tag — right now the CLI passes anything through raw.

File: desktop/src/features/projects/ui/ProjectPullRequestsPanel.tsx [hunk 3]
```diff
@@ -335,18 +329,51 @@ export function PullRequestDetailHeader({
...
+        </span>
+        <span title={formatExactTimestamp(pullRequest.createdAt)}>
+          created {relativeTime(pullRequest.createdAt)}
+        </span>
+        {sourceChannelId ? (
+          <>
+            <span>in</span>
+            <button
+              className="truncate font-medium text-foreground underline-offset-2 hover:underline"
+              onClick={() => void goChannel(sourceChannelId)}
```
Comment (wpfleger96) [inline desktop/src/features/projects/ui/ProjectPullRequestsPanel.tsx]: Heads up that this link reads as verified provenance but the `h` tag is author-claimed and unauthenticated: the relay classifies kind 1617 as global-only, so no membership check ever runs against the claimed channel — any PR author can stamp any channel UUID here, including a private channel they're not in. Name resolution comes from the viewer's own channel list so nothing private leaks, and clicking just navigates, but I'd either note this in a comment or render it visibly as author-claimed rather than as a trusted "in #channel" link.

File: desktop/src/features/projects/lib/projectsViewHelpers.ts [hunk 4]
```diff
@@ -191,21 +191,30 @@ export function formatCreatedDate(createdAt: number) {
...
-    { label: "year", seconds: 365 * 24 * 60 * 60 },
-    { label: "month", seconds: 30 * 24 * 60 * 60 },
-    { label: "week", seconds: 7 * 24 * 60 * 60 },
     { label: "day", seconds: 24 * 60 * 60 },
     { label: "hour", seconds: 60 * 60 },
     { label: "minute", seconds: 60 },
     { label: "second", seconds: 1 },
   ];
 
+  if (elapsedSeconds >= 7 * 24 * 60 * 60) {
```
Comment (wpfleger96) [inline desktop/src/features/projects/lib/projectsViewHelpers.ts:206]: nit: this new >=7-day absolute-date branch has no unit test — `projectsViewHelpers` has no test file at all, and `relativeTime` is now the single time formatter for issues, PRs, commits, and inline comments. A small `.test.mjs` with the year-boundary case would be cheap insurance.

File: desktop/src/features/projects/ui/PullRequestReviewCard.tsx [hunk 5]
```diff
@@ -133,152 +145,183 @@ export function PullRequestReviewCard({
...
-          >
-            Convert to draft
-          </button>
-        </p>
-      ) : null}
-    </div>
+      <Dialog
+        onOpenChange={(open) => {
+          setApproveDialogOpen(open);
+          if (!open && !isApproving) setApprovalSummary("");
```
Comment (wpfleger96) [inline desktop/src/features/projects/ui/PullRequestReviewCard.tsx]: nit: Escape still closes the dialog while the approval mutation is in flight — `onOpenChange` only guards the summary reset, not the close itself. The approval still lands and toasts, but the dialog vanishing mid-flight is a bit misleading. Consider blocking close while `isApproving`.

File: desktop/src/features/projects/ui/ProjectPullRequestFilesChangedPanel.tsx [hunk 6]
```diff
@@ -475,17 +495,27 @@ function DiffPreview({
...
                 diffLineClassName(row.type),
+                isFocused && "bg-primary/10 ring-1 ring-primary/40 ring-inset",
               )}
               data-line={anchor?.line}
               data-path={anchor?.path}
               data-side={anchor?.side}
-              data-testid={anchor ? "project-diff-line" : undefined}
+              data-testid={
+                isFocused
+                  ? "project-diff-focused-line"
```
Comment (wpfleger96) [inline desktop/src/features/projects/ui/ProjectPullRequestFilesChangedPanel.tsx:534]: nit: the `project-diff-focused-line` highlight persists indefinitely after inline-comment navigation — there's no clear-on-interaction. Cosmetic, fine to leave, just flagging.

File: desktop/src/testing/e2eBridge.ts [hunk 7]
```diff
@@ -9644,6 +9645,38 @@ export function maybeInstallE2eTauriMocks() {
...
+          closed: KIND_GIT_STATUS_CLOSED,
+        }[input.status];
+        const event = createMockEvent(
+          kind,
+          "",
+          [
+            ["e", input.pullRequestId, "", "root"],
+            ["a", input.repoAddress],
+            ["p", input.targetOwner],
+            ["p", input.pullRequestAuthor],
```
Comment (wpfleger96) [inline desktop/src/testing/e2eBridge.ts]: nit: this mock always pushes both `p` tags, but the real Rust command dedupes when author == owner. Mock-fidelity only, no behavioral impact on current specs.

File: crates/buzz-acp/src/base_prompt.md [hunk 8]
```diff
@@ -17,10 +17,13 @@ The `buzz` CLI is your primary interface. Auth env vars: `BUZZ_RELAY_URL`, `BUZZ
 | `buzz feed` | `get` |
 | `buzz social` | `publish`, `notes` |
 | `buzz repos` | `create`, `get`, `list` |
+| `buzz pr` | `open`, `update`, `get`, `list`, `status` |
 | `buzz upload` | `file` |
 
 Run `buzz --help` or `buzz <group> --help` for full usage. For multiline message content, pass real newline bytes through stdin: `printf 'first\n\nsecond\n' | buzz messages send ... --content -`. Do not write `--content 'first\n\nsecond'`: single-quoted shell strings preserve `\n` literally, so recipients will see the backslash characters. `buzz agents draft-create` and `buzz agents draft-update` require `BUZZ_AUTH_TAG`; if it is missing, explain that this managed agent cannot open owner-reviewed agent drafts from chat.
 
+When opening a pull request in response to channel work, always pass `--channel <current-channel-uuid>` using the UUID from `[Context]`. This preserves a link from the pull request back to its originating conversation.
```
Comment (wpfleger96) [inline crates/buzz-acp/src/base_prompt.md:25]: Observation, no action needed: agents following this guidance will embed channel UUIDs in globally-readable PR events. UUIDs are opaque and grant no access, but it does disclose private-channel identifiers relay-wide — I'm assuming that's the intended tradeoff for source-channel linking, just want it to be a conscious one.

**Review by wpfleger96 — COMMENTED** [review]
Leaving consolidated review findings from two independent passes at head `b6d9ac880`. The one item I'd fix before merge is the review-decision timestamp ordering (inline on `hooks.ts`) — the rest are minor/nits, inline below. What held up well: the lifecycle status command reuses the existing merge-path trust gates (`project_owner_identity` + `spawn_key_refusal`, rejects the `merged` alias, clamps `created_at`), `canReviewProjectPullRequest` is one gate shared across all three call sites, and the E2E coverage asserts signed-event payloads rather than just UI state.

File: desktop/src/features/projects/hooks.ts (hunk 1 above)
Comment (thomaspblock) [reply desktop/src/features/projects/hooks.ts:496]: 🤖

> the later decision can lose

Fixed in fcc5e2415: request-change decisions now use the same PR-scoped monotonic timestamp allocator as approvals. Added a deterministic same-second approve → request-changes Playwright regression.

**Comment by thomaspblock** [comment]
updated

**Review by wpfleger96 — APPROVED** [review]
